file/file_prefetch_buffer.cc

```diff
@@ -315,17 +315,41 @@ Status FilePrefetchBuffer::PrefetchAsyncInternal(
 
     // Move data from curr_ buffer to third.
     CopyDataToBuffer(curr_, offset, length);
+    // swap the buffers.
```

Comment:
This will still block on line 276. Can we copy data from curr_ buffer before that when the required data overlaps the end of curr_ and into second_, thus emptying curr_ and starting a new async read? That way, while this polls for second_ to complete, the async read to curr_ happens in parallel.